cmd/tailscale/cli/update.go

```diff
@@ -333,6 +333,73 @@ func (up *updater) updateMacSys() error {
 	return errors.New("The 'update' command is not yet implemented on macOS.")
 }
 
+func (up *updater) updateMacAppStore() error {
+	fmt.Println(`This is the macOS App Store version of Tailscale. Updates are managed via the
+App Store, and the program 'softwareupdate'.
+
+See https://tailscale.com/s/unstable-clients to use TestFlight or to install
+the non-App Store version.`)
+
+	cmd := exec.Command("softwareupdate", "--schedule")
+	stdout := bytes.NewBuffer(nil)
+	stderr := bytes.NewBuffer(nil)
+	cmd.Stdout = stdout
+	cmd.Stdout = stderr
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("can't check App Store update schedule: %v", err)
+	}
+	fmt.Println(stdout.String())
+
+	cmd = exec.Command("softwareupdate", "--list")
+	stdout.Reset()
+	stderr.Reset()
+	cmd.Stdout = stdout
+	cmd.Stdout = stderr
+	if err := cmd.Run(); err != nil {
+		fmt.Fprintln(os.Stderr, stdout.String())
+		return fmt.Errorf("can't check App Store for available updates: %v", err)
+	}
+
+	newTailscale := parseSoftwareupdateList(stdout.Bytes())
+	if newTailscale == "" {
+		fmt.Println("no Tailscale update available")
+		return nil
+	}
+
+	if err := up.confirm(strings.Split(newTailscale, "-")[1]); err != nil {
+		return err
+	}
+
+	cmd = exec.Command("sudo", "softwareupdate", "--install", newTailscale)
+	stdout.Reset()
+	stderr.Reset()
+	cmd.Stdout = stdout
+	cmd.Stdout = stderr
+	if err := cmd.Run(); err != nil {
+		fmt.Fprintln(os.Stderr, stdout.String())
+		return fmt.Errorf("can't install App Store update for Tailscale: %v", err)
+	}
+	return nil
+}
+
+// parseSoftwareupdateList searches the output of `softwareupdate --list` on
+// Darwin and returns the matching Tailscale package label. If there is none,
+// returns the empty string.
+//
+// See TestParseSoftwareupdateList for example inputs.
+func parseSoftwareupdateList(stdout []byte) string {
+	re, err := regexp.Compile(`(?m)^\s+\*\s+Label:\s*(Tailscale-\d[\d\.]+)`)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Could not compile regexp: %v", err)
+		return ""
+	}
```

Comment:
you can create a package-level var via `regexp.MustCompile` and catch regexp syntax errors early